library/oneview_id_pools_ipv4_range.py

```diff
@@ -0,0 +1,140 @@
+            - List with ID pools IPV4 Range properties.
+        required: true
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.validateetag
+'''
+
+EXAMPLES = '''
+- name: Ensure that ID pools IPV4 Range is present using the default configuration
+  oneview_id_pools_ipv4_range:
+    config: "{{ config_file_path }}"
+    state: present
+    data:
+      name: 'Test ID pools IPV4 Range'
+
+- name: Ensure that ID pools IPV4 Range is absent
+  oneview_id_pools_ipv4_range:
+    config: "{{ config_file_path }}"
+    state: absent
+    data:
+      name: 'ID pools IPV4 Range'
+'''
+
+RETURN = '''
+id_pools_ipv4_range:
+    description: Has the facts about the OneView ID pools IPV4 Ranges.
+    returned: On state 'present'. Can be null.
+    type: complex
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from module_utils.oneview import OneViewModuleBase, HPOneViewValueError, ResourceComparator
+
+
+class IdPoolsIpv4RangeModule(OneViewModuleBase):
+    MSG_CREATED = 'ID pools IPV4 Range created successfully.'
+    MSG_UPDATED = 'ID pools IPV4 Range updated successfully.'
+    MSG_DELETED = 'ID pools IPV4 Range deleted successfully.'
+    MSG_ALREADY_PRESENT = 'ID pools IPV4 Range is already present.'
+    MSG_ALREADY_ABSENT = 'ID pools IPV4 Range is already absent.'
+    RESOURCE_FACT_NAME = 'id_pools_ipv4_range'
+
+    def __init__(self):
+
+        additional_arg_spec = dict(data=dict(required=True, type='dict'),
+                                   state=dict(
+                                       required=True,
+                                       choices=['present', 'absent']))
+
+        super(IdPoolsIpv4RangeModule, self).__init__(additional_arg_spec=additional_arg_spec,
+                                                     validate_etag_support=True)
+
+        self.resource_client = self.oneview_client.id_pools_ipv4_ranges
+
+    def execute_module(self):
+        resource = None
+        if self.data.get('uri'):
+            resource = self.resource_client.get(self.data.get('uri'))
+        elif self.data.get('subnetUri') and self.data.get('name'):
+            subnet = self.oneview_client.id_pools_ipv4_subnets.get(self.data.get('subnetUri'))
+            for range_uri in subnet['rangeUris']:
+                maybe_resource = self.resource_client.get(range_uri)
+                if maybe_resource['name'] == self.data['name']:
+                    resource = maybe_resource
+                    break
+
+        self.data['type'] = self.data.get('type', 'Range')
+
+        if self.state == 'present':
+            return self.__present(resource)
+        elif self.state == 'absent':
+            return self.resource_absent(resource)
+
+    def __present(self, resource):
+        if not resource:
+            response = self.resource_present(resource, 'id_pools_ipv4_range')
+        else:
+            # Enabled can be only True, False or None. Using not found default 'X' for comparison purposes.
```

Comment:
I'd suggest swapping out to something like:
`Enabled can be: True, False or None`
or
`Enabled can only be True, False or None`

The "can be only" reads a bit weird.